TemplePlus/dispatcher.cpp

```diff
@@ -72,6 +77,40 @@ int DispatcherReplacements::Dispatch54AoE(objHndl handle, DispIO* evtObj, D20Dis
 	return 0;
 }
 
+int DispatcherReplacements::GetNumNaturalAttacks_(objHndl handle) {
+	if (!handle) return 0;
+	auto obj = objSystem->GetObject(handle);
+	int result = 0;
+	int idx = 0;
+	int attacksCount;
+	do
+	{
+		attacksCount = obj->GetInt32(obj_f_critter_attacks_idx, idx);
+		if (attacksCount > 0)
+			result += attacksCount;
+		idx++;
+	} while (idx < 4);
+	return result;
+}
+
+int DispatcherReplacements::GetDamageIdx(objHndl handle, int idx)
+{
+	if (!handle) return 0;
+	auto obj = objSystem->GetObject(handle);
+	int v2 = 0;
+	int result = 0;
+	while (1)
```

Comment:
Likewise, decompiler produces crappy formatting. What seems like an infinite loop is really just a simple for loop.

int __cdecl GetDamageIdx(ObjHndl ObjHnd, int attackIdx){
int attackIdxSum = 0;
for (auto idx = 0; idx < 5; idx++){
    attackIdxSum += obj->GetInt32(obj_f_critter_attacks_idx, idx);
    if ( attackIdxSum > attackIdx )
      return idx;
  }
  return 0; // attackIdx is bigger than the sum of all obj_f_critter_attacks_idx items

Reply:
I wanted to change as little as possible. Thanks for advice. Will fix

Reply:
Also, it looks like I already rewrote this function!

And in critterSys too :)

Reply:
On that note. Since Temple+ already replaces a whole bunch of stuff, you should be extra careful when replacing existing functions. Sometimes doing a replaceFunction won't help because of exactly what happened here: I rewrote this function and called it explicitly in the code. That's my fault really, but that's how it is right now...